### PR TITLE
Fix invalid input errors for Scc Account

### PR DESCRIPTION
**Pull request by nadjaheitmann**

### CODE REVIEW

**Comment by nadjaheitmann** [comment]
It is still possible to add an invalid date while the sync interval is set to 'never'. This is because the input field seems to return `nil` if the date is not valid, which is not caught by our validator. The only solution would be to through an error every time the sync_date field is not set. I would leave it as is for now as it might be more annoying to set a date every time, even if it is not needed.  

We can fix this once we move to react (or to Foreman 2.5, the timedate field changes completely).

File: test/fixtures/models/scc_accounts.yml [hunk 1]
```diff
@@ -14,6 +15,7 @@ two:
   name: twoname
   interval: never
   organization_id: <%= ActiveRecord::FixtureSet.identify(:empty_organization) %>
+  sync_date: 2021-07-05T21:33:49
```
Comment (m-bucher) [inline test/fixtures/models/scc_accounts.yml:18]: Is there a reason we use a different format here than at line 9?

**Review by m-bucher — APPROVED** [review]
:1st_place_medal: for introducing more tests

File: test/fixtures/models/scc_accounts.yml (hunk 1 above)
Comment (nadjaheitmann) [reply test/fixtures/models/scc_accounts.yml:18]: Not really, I just wanted to have some variation. I don't think that it really matters.

Comment (m-bucher) [reply test/fixtures/models/scc_accounts.yml:18]: Fine with me :smiley:

**Comment by nadjaheitmann** [comment]
@m-bucher Thanks for reviewing!

**Pull request merged**